### PR TITLE
Upgrade to SM 60

### PR DESCRIPTION
**Pull request by asajeffrey**
This is the mozjs_sys smup PR. Depends on https://github.com/servo/mozjs/pull/138.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-mozjs/430)

### CODE REVIEW

**Comment by asajeffrey** [comment]
Yay, the Travis Linux job is running, just Travis OSX and Appveyor to go!

**Comment by asajeffrey** [comment]
Squashed.

**Comment by asajeffrey** [comment]
The travis failures are  https://github.com/rust-lang/rust/issues/51699.

File: src/conversions.rs [hunk 1]
```diff
@@ -600,6 +600,7 @@ impl<C: Clone, T: FromJSValConvertible<Config=C>> FromJSValConvertible for Vec<T
         let mut iterator = ForOfIterator {
             cx_: cx,
             iterator: RootedObject::new_unrooted(),
+            nextMethod: RootedValue::new_unrooted(),
             index: ::std::u32::MAX, // NOT_ARRAY
```
Comment (jdm) [inline src/conversions.rs]: On the android cross-compile for mac, this has an additional `__bindgen_padding_0` field.

Comment (asajeffrey) [reply src/conversions.rs]: Oh goody, a `cfg` that depends on both the source and the target. Do you know what its type is?

Comment (jdm) [reply src/conversions.rs]: It's a u32.

Comment (asajeffrey) [reply src/conversions.rs]: I put it in when the target is android, hopefully it doesn't depend on the host OS too.

Comment (jdm) [reply src/conversions.rs]: The `__bindgen_padding_0` was removed again, and its lack is breaking my build when I build the smup Servo branch.

Comment (jdm) [reply src/conversions.rs]: Specifically, 694bc50  should be removed from this pull request to fix this problem.

File: src/jsglue.cpp [hunk 2]
```diff
@@ -442,45 +425,6 @@ class ForwardingProxyHandler : public js::BaseProxyHandler
     }
 };
 
-class ServoDOMVisitor : public JS::ObjectPrivateVisitor {
```
Comment (jdm) [inline src/jsglue.cpp]: Did this get replaced?

File: src/jsglue.cpp [hunk 3]
```diff
@@ -849,18 +804,11 @@ static size_t MallocSizeOf(const void* aPtr)
...
-
-  return JS::AddServoSizeOf(rt, MallocSizeOf, &sdv, sizes);
-}
-
-void
-InitializeMemoryReporter(WantToMeasure wtm){
-  gWantToMeasure = wtm;
+    mozilla::PodZero(sizes);
+    return JS::AddServoSizeOf(cx, MallocSizeOf,
+                              /* ObjectPrivateVisitor = */ nullptr, sizes);
```
Comment (jdm) [inline src/jsglue.cpp]: I'm concerned that we're regressing the DOM measurement.

File: src/rust.rs [hunk 4]
```diff
@@ -121,13 +126,13 @@ impl ToResult for bool {
 thread_local!(static CONTEXT: Cell<*mut JSContext> = Cell::new(ptr::null_mut()));
 
 lazy_static! {
+    static ref PARENT: AtomicPtr<JSRuntime> = AtomicPtr::new(ptr::null_mut());
     static ref OUTSTANDING_RUNTIMES: AtomicUsize = AtomicUsize::new(0);
     static ref SHUT_DOWN: AtomicBool = AtomicBool::new(false);
 }
 
-/// A wrapper for the `JSRuntime` and `JSContext` structures in SpiderMonkey.
+/// A wrapper for the `JSContext` structures in SpiderMonkey.
```
Comment (jdm) [inline src/rust.rs]: Now only a single structure.

File: src/rust.rs [hunk 5]
```diff
@@ -981,8 +990,11 @@ pub unsafe fn define_properties(cx: *mut JSContext, obj: HandleObject,
                                 -> Result<(), ()> {
     assert!({
         match properties.last() {
-            Some(&JSPropertySpec { name, flags, getter, setter }) => {
-                name.is_null() && flags == 0 && getter.is_zeroed() && setter.is_zeroed()
+            Some(spec) => {
+                let zero: JSPropertySpec = mem::zeroed();
+                let zero: [usize; 6] = mem::transmute(zero);
+                let words: &[usize; 6] = mem::transmute(spec);
+                *words == zero
```
Comment (jdm) [inline src/rust.rs]: Can we use `spec.is_zeroed()` instead?

File: src/jsglue.cpp (hunk 2 above)
Comment (asajeffrey) [reply src/jsglue.cpp]: Hmm, I'll have to have a look.

File: src/jsglue.cpp (hunk 3 above)
Comment (asajeffrey) [reply src/jsglue.cpp]: A fair concern. I;ll have a look to see if there's a replacement for ServoDOMVisitor.

File: src/rust.rs (hunk 4 above)
Comment (asajeffrey) [reply src/rust.rs]: Fixed.

File: src/rust.rs (hunk 5 above)
Comment (asajeffrey) [reply src/rust.rs]: Fixed.

File: src/jsglue.cpp (hunk 2 above)
Comment (asajeffrey) [reply src/jsglue.cpp]: Restored it.

**Comment by asajeffrey** [comment]
https://mozilla.logbot.info/servo/20180815#c15173056 @bors-servo r=jdm

**Comment by asajeffrey** [comment]
@bors-servo retry
 * travis timed out

**Comment by asajeffrey** [comment]
@bors-servo retry
 * homu, what do you mean, status-travis, it's green!

**Pull request merged**